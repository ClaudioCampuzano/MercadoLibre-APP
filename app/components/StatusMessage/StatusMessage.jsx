import React from 'react';
import Image from 'next/image';
import { Error, Spinner } from '@/public';
import styles from './StatusMessage.module.scss';

const StatusMessage = ({ status }) => {
  if (status === 'loading')
    return (
      <div className={styles.container}>
        <Image src={Spinner} alt="Loading" width="50" height="50" />
        <span>Cargando...</span>
      </div>
    );

  if (status === 'error')
    return (
      <div className={styles.container}>
        <Image src={Error} alt="Error" width="50" height="50" />
        <span>Error, intente nuevamente</span>
      </div>
    );

  return undefined;
};

export { StatusMessage };
