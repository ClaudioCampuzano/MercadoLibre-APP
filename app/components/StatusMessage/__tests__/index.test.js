import React from 'react';

import { render } from '@testing-library/react';

import { StatusMessage } from '../StatusMessage';

const componentRender = (props = { status: 'loading' }) => {
  return render(<StatusMessage {...props} />);
};

describe('StatusMessage', () => {
  it('renders', () => {
    const { container } = componentRender();

    expect(container).toMatchSnapshot();
  });
});
