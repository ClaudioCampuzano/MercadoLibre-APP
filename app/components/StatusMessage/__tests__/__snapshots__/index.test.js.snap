// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMessage renders 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="Loading"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
    <span>
      Cargando...
    </span>
  </div>
</div>
`;
